File: Java/subscription/GetSubscriptionById.java

package examples.subscription;

import me.pagar.api.PagarmeCoreApiClient;
import me.pagar.api.controllers.SubscriptionsController;
import me.pagar.api.http.client.*;
import me.pagar.api.models.GetSubscriptionResponse;

public class GetSubscriptionById {

    public static void main(String[] args) {

        // Configuration parameters and credentials
        String basicAuthUserName = "basicAuthUserName"; // The username to use with basic authentication
        String basicAuthPassword = "basicAuthPassword"; // The password to use with basic authentication

        PagarmeCoreApiClient client = new PagarmeCoreApiClient(basicAuthUserName, basicAuthPassword);

        SubscriptionsController subscriptions_controller = new SubscriptionsController();

        String subscriptionId = "sub_vJZV7aef35tQXndo";

        subscriptions_controller.getSubscriptionAsync(subscriptionId, new APICallBack<GetSubscriptionResponse>() {
            public void onSuccess(HttpContext context, GetSubscriptionResponse response) {
                System.out.println("Subscription found !");
                System.out.println("Status response: " + context.getResponse().getStatusCode());
                System.out.println("Subscription id: " + response.getId());
            }

            public void onFailure(HttpContext context, Throwable error) {

                System.out.println("Status response: " + context.getResponse().getStatusCode());
                System.out.println(error.getMessage());
                error.printStackTrace();
            }
        });
    }

}

File: Python/subscription/get_subscription_by_id.py
from pagarmecoreapi.pagarmecoreapi_client import PagarmecoreapiClient
from pagarmecoreapi.controllers import *
from pagarmecoreapi.exceptions.error_exception import *

# Configuration parameters and credentials
basic_auth_user_name = 'basic_auth_user_name' # The username to use with basic authentication
basic_auth_password = 'basic_auth_password' # The password to use with basic authentication

PagarmecoreapiClient(basic_auth_user_name, basic_auth_password)

subscriptions_controller = subscriptions_controller.SubscriptionsController()

subscription_Id = "sub_2EvZ8GdFYZhXkbe4"

try:
    result = subscriptions_controller.get_subscription(subscription_Id)
    assert result is not None
    assert result.id == subscription_Id
    print("Subscription id: ", result.id)
except ErrorException as ex:
    print(ex.message)
    print("Errors: ", ex.errors)
except Exception as ex:
    raise ex
    